### PR TITLE
feat(atomic): expose & document Stencil decorators

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/KIT-1258

### CODE REVIEW

File: packages/atomic/README.md [hunk 1]
```diff
@@ -86,24 +50,41 @@ npm run cypress:test
...
+### Stencil decorators
+
+To build custom Atomic components, it is highly recommended to use the same toolchain that we used, [Stencil](https://stenciljs.com/).
+We provide decorators which are used internally with most of our components.
 
-The `InitializeBindings` is an utility that automatically fetches the `bindings` from the parent `atomic-search-interface` component. This decorator should be applied to the `bindings` property directly.
+## InitializeBindings & BindStateToController decorators
 
-_Important_ In order for a component using this decorator to render properly, it should have an internal state bound to one of the property from `bindings`. This is possible by using either the `BindStateToController`decorator.
+`InitializeBindings` is an utility that automatically fetches the `bindings` from the parent `atomic-search-interface` or `atomic-external` component. This decorator has to be applied a property named `bindings`.
```
Comment (samisayegh) [inline packages/atomic/README.md:60]: `InitializeBindings is an utility` -> `InitializeBindings is a utility

`has to be applied a property named` -> `has to be applied to a property named`

**Review by samisayegh — APPROVED** [review]

File: packages/atomic/src/components/result-template-components/result-template-decorators.tsx [hunk 2]
```diff
@@ -10,6 +10,17 @@ export class MissingResultParentError extends Error {
   }
 }
 
+/**
+ * A [StencilJS property decorator](https://stenciljs.com/) to be used for result template components.
+ * Allows the Stencil component to resolve the curent result in which the component is being rendered.
```
[the remark this answers is not in the document]
Comment (ThibodeauJF) [reply packages/atomic/src/components/result-template-components/result-template-decorators.tsx:15]: Yes! in the readme it should be more clear

File: packages/atomic/src/components/result-template-components/result-template-decorators.tsx [hunk 3]
```diff
@@ -10,6 +10,17 @@ export class MissingResultParentError extends Error {
...
+ *
+ * In the case of
```
[the remark this answers is not in the document]
Comment (ThibodeauJF) [reply packages/atomic/src/components/result-template-components/result-template-decorators.tsx:17]: I'll remove this :D
```suggestion
```

File: packages/atomic/src/components/result-template-components/result-template-decorators.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (dlutzCoveo) [reply packages/atomic/src/components/result-template-components/result-template-decorators.tsx:15]:
```suggestion
 * This allows the Stencil component to fetch the current result from its rendered parent, the `atomic-result` component.
```

**Pull request merged**